src/phoenix/trace/langchain/tracer.py

```diff
@@ -170,7 +195,50 @@ def _retrieval_documents(
     ]
 
 
-class OpenInferenceTracer(Tracer, BaseTracer):
+class BaseTracerWithChatModelSupport(BaseTracer):
+    """
+    BaseTracer class with support for chat models.
+
+    LangChain's BaseTracer class does not implement hooks for chat models and hence does not record
+    data such as the list of messages that were passed to the chat model. This class implements
+    those hooks.
+
+    For reference, see https://github.com/langchain-ai/langchain/pull/4499.
+    """
+
+    def on_chat_model_start(
```

Comment:
why not add the method directly to OpenInferenceTracer? wouldn't it be simpler that way?

Reply:
Addressed in [648aa51](https://github.com/Arize-ai/phoenix/pull/1410/commits/648aa51e5379854eae7ff51befd790ebe2defaef)